tools/ci_build/github/azure-pipelines/templates/jobs/download_linux_qnn_sdk.yml

```diff
@@ -16,6 +16,22 @@ steps:
       echo $(QnnSDKRootDir)
     displayName: 'Print QnnSDKRootDir after downloading QNN SDK'
 
+  - script: |
+      set -x
+      sdk_file="$(QnnSDKRootDir)/sdk.yaml"
+      qnn_sdk_version=$(grep '^version:' "$sdk_file" | head -n 1 | cut -d':' -f2 | xargs | cut -d'.' -f1-3)
+      if [[ -n "$qnn_sdk_version" ]]; then
+        echo "##vso[task.setvariable variable=QnnSDKVersionYaml;isOutput=true]$qnn_sdk_version"
+      else
+        echo "Version not found in sdk.yaml."
+        exit 1
+      fi
+    displayName: Set_QnnSDKVersionYaml based on sdk.yaml
+    name: Set_QnnSDKVersionYaml
+  - bash: |
```

Comment:
I saw the trailing quote when I was testing the syntax in sample pipeline, didn't know it was because of set -x. That's good to know

Reply:
So it clashes with the parameter name in download_linux_qnn_sdk.yml?

It looks like in general there's a QnnSdk value that is set at a high level in multiple pipelines.
We pass that as QnnSdkVersion when running download_linux_qnn_sdk.yaml, which sets another variable with the version number. Which seems like it equates to 3 different names defining the same thing.

If in the other CIs we're defining QnnSdk at a high level (which is painful to update) should we follow that pattern here? That would mean the download yaml can validate the version matches (not sure that is necessary given we're specifying the download url), but doesn't need to set a variable as QnnSdk should be used. 

Alternatively, would it be nicer if you don't need the high level QnnSdk unless overriding the default version? If we did that, maybe the parameter in download_linux_qnn_sdk.yml could be called 'QnnSdkVersionToDownload', and it could set a variable called 'QnnSdkVersion' which can be used everywhere else in the CI configs. If we want to take this approach we should check the QNN EP folks have no objections.

Reply:
@skottmckay  "That would mean the download yaml can validate the version matches (not sure that is necessary given we're specifying the download url), but doesn't need to set a variable as QnnSdk should be used_" - you had mentioned in the previous PR to not use the  QnnSdk defined at a high level as it may get build information (2.27.0.240926) and we will not get the correct version number- https://github.com/microsoft/onnxruntime/pull/22296#discussion_r1785646198

"Alternatively, would it be nicer if you don't need the high level QnnSdk unless overriding the default version? If we did that, maybe the parameter in download_linux_qnn_sdk.yml could be called 'QnnSdkVersionToDownload', and it could set a variable called 'QnnSdkVersion' which can be used everywhere else in the CI configs" - I believe this is how it's works currentlya. The pipeline parameter is optional and used to override the default version. Since its a parameter and not a variable, it's not modified during run and so passed again as parameter in whichever yaml it's used

If we are okay using the version used to download the SDK, we can get QnnSdkVersion from QnnSdkVersionToDownload (updated name from QnnSDKVersion) param in onnxruntime-android-qnn build and test stage (no need to use stage variables in download_linux_qnn_sdk.yml)

Reply:
Using the version used to download the SDK is fine. 

I would prefer we had a setup where you only need to override the default if necessary as it simplifies changes when we update the QNN SDK version (update the default instead of multiple CI configs) but we can do that separately.